In [88]:
import pandas as pd
import numpy as np

In [89]:
from sklearn.metrics import mean_squared_error

In [90]:
import os

In [91]:
y_true = pd.read_csv('datasets/test_set.csv')['cpi_pct'].to_numpy()

In [92]:
prediction = []
for file_name in os.listdir('predictions/'):
    
    if file_name[:3] == 'pca':
        continue
    
    y_pred = pd.read_csv(f'predictions/{file_name}', index_col=0).to_numpy()
    
    name = file_name.rsplit('.', 1)[0]
    rmse =  mean_squared_error(y_true, y_pred, squared=False)
    print(name, "rmse:", round(rmse, 7))
    prediction.append({'model': name, 'rmse:': rmse})

benchmark1_mean rmse: 0.0032184
benchmark2_prev rmse: 0.0026672
benchmark3_rf rmse: 0.0020702
bma rmse: 0.0020429
lasso_bic rmse: 0.0021015
lasso_cv rmse: 0.0021016
lasso_cvshuffled rmse: 0.0019987
lasso_ebic rmse: 0.0021058
ols rmse: 0.0030478
ridge_cv rmse: 0.0021375


In [95]:
!pip install Jinja2

  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)


$$
\\begin{tabular}{llr}\n\\toprule\n & model & rmse: \\\\\n\\midrule\n0 & benchmark1_mean & 0.003218 \\\\\n1 & benchmark2_prev & 0.002667 \\\\\n2 & benchmark3_rf & 0.002070 \\\\\n3 & bma & 0.002043 \\\\\n4 & lasso_bic & 0.002101 \\\\\n5 & lasso_cv & 0.002102 \\\\\n6 & lasso_cvshuffled & 0.001999 \\\\\n7 & lasso_ebic & 0.002106 \\\\\n8 & ols & 0.003048 \\\\\n9 & ridge_cv & 0.002137 \\\\\n\\bottomrule\n\\end{tabular}\n
$$

In [98]:
pd.DataFrame(prediction).to_latex()

'\\begin{tabular}{llr}\n\\toprule\n & model & rmse: \\\\\n\\midrule\n0 & benchmark1_mean & 0.003218 \\\\\n1 & benchmark2_prev & 0.002667 \\\\\n2 & benchmark3_rf & 0.002070 \\\\\n3 & bma & 0.002043 \\\\\n4 & lasso_bic & 0.002101 \\\\\n5 & lasso_cv & 0.002102 \\\\\n6 & lasso_cvshuffled & 0.001999 \\\\\n7 & lasso_ebic & 0.002106 \\\\\n8 & ols & 0.003048 \\\\\n9 & ridge_cv & 0.002137 \\\\\n\\bottomrule\n\\end{tabular}\n'

2000-2023 model:

- benchmark1_mean rmse: 0.0026948
- benchmark2_prev rmse: 0.0034082
- bma rmse: 0.002297
- lasso_bic rmse: 0.0023243
- lasso_ebic rmse: 0.0023711
- lasso_cv rmse: 0.0027338
- lasso_cvshuffled rmse: 0.0027338
- ols rmse: 0.0279417
- ridge_cv rmse: 0.0024787

1970-2023 model:

- benchmark1_mean rmse: 0.0032184
- benchmark2_prev rmse: 0.0026672
- benchmark3_rf rmse: 0.0020702
- bma rmse: 0.0020429
- lasso_bic rmse: 0.0021015
- lasso_cv rmse: 0.0021016
- lasso_cvshuffled rmse: 0.0019987
- lasso_ebic rmse: 0.0021058
- ols rmse: 0.0030478
- ridge_cv rmse: 0.0021375